### PR TITLE
Merge datasets with yolo_io

### LINKED ISSUE #7: Data Merging
## Description
- Create a script to merge different YOLO-formatted datasets.

### Justification
- YOLO trained on Isaac data seemed to perform worse than when trained on Godot data. 
- The discrepancy might be related to variance caused by new backgrounds. This can be remedied by using different sources of synthetic data.

### PR DESCRIPTION
**Pull request by MichaelKrinsky**
## Description
- Created merge_datasets script and updated yolo_to_yolo functions.

## Issues
- Closes #7

### CODE REVIEW

**Comment by EricPedley** [comment]
lgtm but I'll let Minh take a look before merging. In the meantime can you use this to create a merged Isaac Sim / Godot dataset and start training yolov8n on it?

**Comment by MinhxNguyen7** [comment]
I'll look at the code, but I won't merge it until we at least see that it works to produce one dataset combo.

**Review by MinhxNguyen7 — APPROVED** [review]
LGTM! I just refactored it a bit

**Pull request merged**
